prod/progress_bar/gadget.xml

```diff
@@ -0,0 +1,396 @@
+            width: 50%;
+            height: 50%;
+            background: #ffffff;
+            border-radius: 100% 0 0 0;
+            position: absolute;
+            top: 0;
+            left: 0;
+            content: '';
+        }
+        .loader:after {
+            background: #5050d2;
+            width: 75%;
+            height: 75%;
+            border-radius: 50%;
+            content: '';
+            margin: auto;
+            position: absolute;
+            top: 0;
+            left: 0;
+            bottom: 0;
+            right: 0;
+        }
+    </style>
+    
+</head>
+<body>
+    <ul class="panel view-select clearfix">
+        <li class="view-teacher">__MSG_teacher_view__</li>
+        <li class="view-student active">__MSG_student_view__</li>
+    </ul>
+
+    <div class="panel slider-content">
+        <h3 class="title">__MSG_my_progress__:</h3> 
+        <div id="progressSlider"></div>
+    </div>
+
+    <div class="panel teacher-content">
+    &nbsp;__MSG_table_description__: <br><br>
+        <table class="student-progress">
+            <thead>
+                <tr>
+                    <th>__MSG_student_name__</th>
+                    <th>__MSG_progress__</th>
+                    <th>__MSG_modified__ <span class="refresh-button">&#10227;</span></th>
+                </tr>
+            </thead>
+            <tbody>
+            </tbody>
+        </table>
+    </div>
+
+    <div class="panel loading active">
+        <div class="loader"></div>
+    </div>
+    <script>
+        $(document).ready(function() {
+            var spinTimeout;
+            function AnimateRotate(angle,repeat) {
+                var duration = 1400;
+                spinTimeout = setTimeout(function() {
+                    if(repeat && repeat == "infinite") {
+                        AnimateRotate(angle,repeat);
+                    } else if ( repeat && repeat > 1) {
+                        AnimateRotate(angle, repeat-1);
+                    }
+                },duration)    
+                var $elem = $('.loader');
+
+                $({deg: 0}).animate({deg: angle}, {
+                    duration: duration,
+                    easing: 'linear',
+                    step: function(now) {
+                        $elem.css({
+                            'transform': 'rotate('+ now +'deg)'
+                        });
+                    }
+                });
+            }
+            AnimateRotate(360,"infinite");
+
+            var teacherViewLoaded = false;
+            $('.view-select .view-teacher').click(function() {
+                if (!$(this).hasClass('active')) {
+                    $('.view-teacher, .teacher-content').addClass('active');
+                    $('.view-student, .slider-content').removeClass('active');
+                    if (!teacherViewLoaded) {
+                        $('table tbody').empty();
+                        $('.teacher-content').append('<p></p><div class="loader"></div>');
```

Comment:
I wouldn't add these `<p></p>` elements otherwise they will keep building up each time the loader is displayed. (You're only calling `$('.loader').remove();`)